kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -86,13 +102,52 @@ private TableConfig(
      * @return the value of the table property
      */
     public T fromMetadata(Metadata metadata) {
-        T value = fromString.apply(metadata.getConfiguration().getOrDefault(key, defaultValue));
-        if (!validator.test(value)) {
-            throw new IllegalArgumentException(
-                    String.format("Invalid value for table property '%s': '%s'. %s",
-                            key, value, helpMessage));
+        String value = metadata.getConfiguration().getOrDefault(key, defaultValue);
+        validate(value);
+        return fromString.apply(value);
+    }
+
+    /**
+     * Returns the key of the table property.
+     *
+     * @return the key of the table property
+     */
+    public String getKey() {
+        return key;
+    }
+
+    /**
+     * Validates that the given properties have the delta prefix in the key name, and they are in
+     * the set of valid properties
+     *
+     * @param configurations the properties to validate
+     * @throws InvalidConfigurationValueException if any of the properties are invalid
+     * @throws UnknownConfigurationException if any of the properties are unknown
+     */
+    public static Map<String, String> validateProperties(Map<String, String> configurations) {
+        Map<String, String> validatedConfigurations = new HashMap<>();
+        for (Map.Entry<String, String> kv : configurations.entrySet()) {
+            String key = kv.getKey().toLowerCase(Locale.ROOT);
+            String value = kv.getValue();
+            if (key.startsWith("delta.")) {
+                TableConfig tableConfig = validProperties.get(key);
+                if (tableConfig != null) {
+                    tableConfig.validate(value);
+                    validatedConfigurations.put(tableConfig.getKey(), value);
```

Comment:
nit: add a comment saying that we want to store the case of the property name defined in TableConfig.